src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -743,9 +743,22 @@ static inline JSC::EncodedJSValue jsBufferConstructorFunction_concatBody(JSC::JS
     }
 
     if (callFrame->argumentCount() > 1) {
-        auto byteLengthValue = callFrame->uncheckedArgument(1);
-        byteLength = std::min(byteLength, byteLengthValue.toTypedArrayIndex(lexicalGlobalObject, "totalLength must be a valid number"_s));
-        RETURN_IF_EXCEPTION(throwScope, {});
+        auto totalLengthValue = callFrame->uncheckedArgument(1);
+        if (!totalLengthValue.isUndefined()) {
+            if (UNLIKELY(!totalLengthValue.isNumber())) {
+                throwTypeError(lexicalGlobalObject, throwScope, "totalLength must be a valid number"_s);
+                return JSValue::encode(jsUndefined());
+            }
+
+            auto totalLength = totalLengthValue.toInt32(lexicalGlobalObject);
```

Comment:
Int32? Are you sure it shouldn’t be an Int52? Since you might want a buffer larger than 2 GB?

Reply:
Note that toTypedArrayIndex returns a size_t

Reply:
😱 fixed.

Reply:
ah we should fix tghat